services/intersection-api/api/src/main/java/us/dot/its/jpo/ode/api/accessors/assessments/ConnectionOfTravelAssessment/ConnectionOfTravelAssessmentRepositoryImpl.java

```diff
@@ -1,74 +1,104 @@
-        }
-
-        query.addCriteria(Criteria.where("assessmentGeneratedAt").gte(startTimeDate).lte(endTimeDate));
-
-        if (latest) {
-            query.with(Sort.by(Sort.Direction.DESC, "assessmentGeneratedAt"));
-            query.limit(1);
-        } else {
-            query.limit(props.getMaximumResponseSize());
-        }
-        return query;
-    }
-
-    public long getQueryResultCount(Query query) {
-        return mongoTemplate.count(query, ConnectionOfTravelAssessment.class, collectionName);
+    /**
+     * Get a page representing the count of data for a given intersectionID,
+     * startTime, and endTime
+     *
+     * @param intersectionID the intersection ID to query by, if null will not be
+     *                       applied
+     * @param startTime      the start time to query by, if null will not be applied
+     * @param endTime        the end time to query by, if null will not be applied
+     * @param pageable       the pageable object to use for pagination
+     * @return the paginated data that matches the given criteria
+     */
+    public long count(
+            Integer intersectionID,
+            Long startTime,
+            Long endTime,
+            @Nullable Pageable pageable) {
+        Query query = new IntersectionCriteria()
+                .whereOptional(INTERSECTION_ID_FIELD, intersectionID)
+                .withinTimeWindow(DATE_FIELD, startTime, endTime)
+                .toQuery(pageable);
+        return mongoTemplate.count(query, collectionName);
     }
 
-    // Removes the result limit on a count query.
-    public long getQueryFullCount(Query query) {
-        int limit = query.getLimit();
-        query.limit(-1);
-        long count = mongoTemplate.count(query, ConnectionOfTravelAssessment.class, collectionName);
-        query.limit(limit);
-        return count;
+    /**
+     * Get a page containing the single most recent record for a given
+     * intersectionID, startTime, and endTime
+     *
+     * @param intersectionID the intersection ID to query by, if null will not be
+     *                       applied
+     * @param startTime      the start time to query by, if null will not be applied
+     * @param endTime        the end time to query by, if null will not be applied
+     * @param pageable       the pageable object to use for pagination
+     * @return the paginated data that matches the given criteria
+     */
+    public Page<ConnectionOfTravelAssessment> findLatest(
+            Integer intersectionID,
+            Long startTime,
+            Long endTime) {
+        Query query = new IntersectionCriteria()
+                .whereOptional(INTERSECTION_ID_FIELD, intersectionID)
+                .withinTimeWindow(DATE_FIELD, startTime, endTime)
+                .toQuery();
+        Sort sort = Sort.by(Sort.Direction.DESC, DATE_FIELD);
+        return wrapSingleResultWithPage(
+                mongoTemplate.findOne(
+                        query.with(sort),
```

Comment:
suggestion(non-blocking): keep similar operations together to make it easier for maintainers to understand how the Query is modified/constructed.
```suggestion
        var criteria = new IntersectionCriteria()
                .whereOptional(INTERSECTION_ID_FIELD, intersectionID)
                .withinTimeWindow(DATE_FIELD, startTime, endTime);
        Sort sort = Sort.by(Sort.Direction.DESC, DATE_FIELD);
        var query = new Query(criteria).with(sort);
        
        return wrapSingleResultWithPage(
                mongoTemplate.findOne(
                        query,
                        ConnectionOfTravelNotification.class,
                        collectionName)
        );
```